extensions/websockets-next/server/runtime/src/main/java/io/quarkus/websockets/next/OnError.java

```diff
@@ -0,0 +1,35 @@
+package io.quarkus.websockets.next;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import io.quarkus.websockets.next.WebSocketConnection.HandshakeRequest;
+import io.smallrye.common.annotation.Experimental;
+
+/**
+ * A {@link WebSocket} endpoint method annotated with this annotation is invoked when an error occurs.
+ * <p>
+ * The method must accept exactly one "error" parameter, i.e. a parameter that is assignable from {@link java.lang.Throwable}.
+ * The method may also accept the following parameters:
+ * <ul>
+ * <li>{@link WebSocketConnection}</li>
+ * <li>{@link HandshakeRequest}</li>
+ * <li>{@link String} parameters annotated with {@link PathParam}</li>
+ * </ul>
+ * <p>
+ * An endpoint may declare multiple methods annotated with this annotation. However, each method must declare a unique error
+ * parameter.
```

Comment:
I think it makes sense to mention the general algorithm used for matching the exception handler

Reply:
Yep, as I said we'll add docs in a follow-up PR but it would not hurt to add this info in the javadoc now...